packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -322,6 +322,31 @@ export interface MfaSecondFactor {
   readonly otp: boolean;
 }
 
+/**
+ * UserPooolAddOns settings for the user pool.
+ * @see https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-userpooladdons.html
+ */
+export interface UserPoolAddOnsSettings {
```

Comment:
Drop this structure and put the `advancedSecurityMode` directly into `UserPoolProps`.